### PR TITLE
Tao 849 form field validation mechanism

### PR DESCRIPTION
**Pull request by hutnikau**

### CODE REVIEW

**Comment by krampstudio** [comment]
As said previously I cannot accept the PR with qtip. So please, remove qtip and the qtip highighter (in fact keep it somewhere).
Then in a separate task we will replace tooltipster by qtip (I will add it in the backlog and argue to have it planned in the next sprint);

**Comment by hutnikau** [comment]
I removed qtip. Please review it again.

File: views/templates/client_config.tpl [hunk 1]
```diff
@@ -53,7 +53,7 @@ require.config({
         'tooltipster'       : 'lib/tooltipster/jquery.tooltipster',
         'nouislider'        : 'lib/sliders/jquery.nouislider',
         'jquery.trunc'		: 'lib/jquery.badonkatrunc',
-        'jquery.fileDownload'  : 'lib/jquery.fileDownload',
+        'jquery.fileDownload'  : 'lib/jquery.fileDownload'
```
Comment (krampstudio) [inline views/templates/client_config.tpl]: the trailing comma is missing

Comment (hutnikau) [inline views/templates/client_config.tpl]: fixed

**Comment by krampstudio** [comment]
The unit is in error on : 

```
Testing http://127.0.0.1:8082/tao/views/js/test/ui/formValidator/highlighters/test.html ...F
>> highlight (tooltip)
>> Message: Unhighlighted tooltip is removed
>> Actual: 1
>> Expected: 0
>> at http://127.0.0.1:8082/tao/views/js/lib/qunit/qunit.js:556
>>   at http://127.0.0.1:8082/tao/views/js/test/ui/formValidator/highlighters/test.js:66
>>   at http://127.0.0.1:8082/tao/views/js/lib/qunit/qunit.js:203
>>   at http://127.0.0.1:8082/tao/views/js/lib/qunit/qunit.js:361
>>   at process (http://127.0.0.1:8082/tao/views/js/lib/qunit/qunit.js:1453)
>>   at http://127.0.0.1:8082/tao/views/js/lib/qunit/qunit.js:479
```

**Comment by krampstudio** [comment]
A last thing: your tests doesn't use the code coverage and there's a method `bindValues` but you should use parameterized tests instead (there's a qunit plugin in tao for that). (tests are the only space where you can repeat, never add something that can change the tested behavior)

**Comment by hutnikau** [comment]
I have fixed QUnit tests. Please review it.

File: views/js/test/ui/formValidator/test.js [hunk 2]
```diff
@@ -33,60 +39,70 @@ define(['lodash', 'jquery', 'ui/formValidator/formValidator'], function(_, $, Fo
...
-    QUnit.test('Highlight/unhighlight fields', function(assert) {
-        var validator = new FormValidator({
-            container : $('#form_1'),
-            event : 'change',
-            selector : fieldSelector
+    QUnit.cases(casesValid)
+        .test('validate valid form', function(data, assert) {
+            QUnit.expect(1);
+
+            bindValues(data, $('#form_1'));
```
Comment (krampstudio) [inline views/js/test/ui/formValidator/test.js]: unit test is the only place you can duplicate code. Please remove the `bindValues` function and hard code the binding in your test case. The principle is to not introduce side effect in any way (or the least). @llecaque can give you more hints on that subject if needed

File: views/js/test/ui/formValidator/test.js [hunk 3]
```diff
@@ -1,92 +1,112 @@
...
+            QUnit.expect(1);
 
-        var validator = new FormValidator({
-            container : $('#form_1'),
-            event : 'change',
-            selector : fieldSelector
-        });
+            _.forEach(data, function (value, selector) {
+                $(selector).val(value);
+            });
```
Comment (hutnikau) [inline views/js/test/ui/formValidator/test.js]: @krampstudio Can I do it in this manner (using _.forEach loop)? O I should insert each value one by one?

Comment (krampstudio) [inline views/js/test/ui/formValidator/test.js]: a forEach is fine

**Pull request merged**